src/pages/permissions/panels/GroupSelection.tsx

```diff
@@ -0,0 +1,151 @@
+import { DependencyList, FC, useState } from "react";
+import PermissionGroupsFilter from "../PermissionGroupsFilter";
+import ScrollableContainer from "components/ScrollableContainer";
+import { LxdGroup } from "types/permissions";
+import { EmptyState, Icon } from "@canonical/react-components";
+import ScrollableTable from "components/ScrollableTable";
+import SelectableMainTable from "components/SelectableMainTable";
+import { Link } from "react-router-dom";
+import { pluralize } from "util/instanceBulkActions";
+
+interface Props {
+  groups: LxdGroup[];
+  modifiedGroups: Set<string>;
+  selectedItemName: string;
+  selectedGroups: Set<string>;
+  setSelectedGroups: (val: string[], isUnselectAll?: boolean) => void;
+  indeterminateGroups?: Set<string>;
+  toggleGroup?: (rowName: string) => void;
+  scrollDependencies: DependencyList;
+}
+
+const GroupSelection: FC<Props> = ({
+  groups,
+  modifiedGroups,
+  selectedItemName,
+  selectedGroups,
+  indeterminateGroups,
+  setSelectedGroups,
+  toggleGroup,
+  scrollDependencies,
+}) => {
+  const [search, setSearch] = useState("");
+
+  const headers = [
+    {
+      content: "Group name",
+      sortKey: "name",
+      className: "name",
+      role: "rowheader",
+    },
+    {
+      content: "Description",
+      sortKey: "description",
+      className: "description",
+      role: "rowheader",
+    },
+    {
+      content: "",
+      role: "rowheader",
+      "aria-label": "Modified status",
+      className: "modified-status",
+    },
+  ];
+
+  const filteredGroups = groups.filter((group) => {
+    return group.name.toLowerCase().includes(search) || !search;
+  });
+
+  const rows = filteredGroups.map((group) => {
+    const groupAdded =
+      selectedGroups.has(group.name) && modifiedGroups.has(group.name);
+    const groupRemoved =
+      !selectedGroups.has(group.name) && modifiedGroups.has(group.name);
+
+    const modifiedTitle = groupAdded
+      ? `Group will be added to all selected ${pluralize(selectedItemName, 2)}`
+      : groupRemoved
+        ? `Group will be removed from all selected ${pluralize(selectedItemName, 2)}`
+        : "";
```

Comment:
The assumption we always have 2 or more is not true, we can have just 1 selected item at hand, then It should probably say "Group will be added/removed to the selected foo" instead of  "...all seleced foos".